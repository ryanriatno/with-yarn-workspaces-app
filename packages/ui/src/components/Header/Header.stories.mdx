import {Meta, Story, Preview, Props} from '@storybook/addon-docs/blocks'

import Header from './index'

<Meta title="Components/Header" component={Header} />

# Header

This is the `Header`

```jsx
import { Header } from 'header'
```

<Preview withSource="open">
  <Story name="Header">
    <div className="relative">
      <Header onClick={() => console.log('test')}>
        <span className="font-semibold">Masuk</span>
      </Header>
    </div>
  </Story>
</Preview>