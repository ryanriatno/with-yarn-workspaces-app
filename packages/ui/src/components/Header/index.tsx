import React, { ReactNode } from 'react'
import './styles.css'
interface Props {
  onClick: any
  children?: ReactNode
}

export const Header: React.FC<Props> = (props) => {
  return (
    <header className="fixed flex flex-row min-w-full bg-clrPrimary h-header z-10">
      <div className="flex items-center w-480 px-3 mx-auto text-white">
        <button
          className="mr-5 leading-none border-none focus:outline-none active:outline-none"
          onClick={props.onClick}
        >
          <span className="icon icon-arrow-left"></span>
        </button>
        <div>{props.children}</div>
      </div>
    </header>
  )
}
export default Header