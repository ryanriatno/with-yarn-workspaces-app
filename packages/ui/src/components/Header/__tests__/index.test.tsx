import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import Header from '../index'

describe("<Header /> test case", () => {
  test("Render Header without error", () => {
    const { container } = render(<Header onClick={() => { return }} />);
    expect(container).toMatchSnapshot();
  });
});