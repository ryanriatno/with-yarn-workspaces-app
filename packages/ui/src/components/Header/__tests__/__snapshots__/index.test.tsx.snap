// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> test case Render Header without error 1`] = `
<div>
  <header
    class="fixed flex flex-row min-w-full bg-clrPrimary h-header z-10"
  >
    <div
      class="flex items-center w-480 px-3 mx-auto text-white"
    >
      <button
        class="mr-5 leading-none border-none focus:outline-none active:outline-none"
      >
        <span
          class="icon icon-arrow-left"
        />
      </button>
      <div />
    </div>
  </header>
</div>
`;
